pkg/proxy/ipvs/proxier.go

```diff
@@ -1328,17 +1328,22 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 }
 
 func (proxier *Proxier) cleanLegacyService(atciveServices map[string]bool, currentServices map[string]*utilipvs.VirtualServer) {
+	unbindIPAddr := map[string]bool{}
 	for cS := range currentServices {
 		if !atciveServices[cS] {
 			svc := currentServices[cS]
 			err := proxier.ipvs.DeleteVirtualServer(svc)
 			if err != nil {
 				glog.Errorf("Failed to delete service, error: %v", err)
 			}
-			err = proxier.ipvs.UnbindVirtualServerAddress(svc, DefaultDummyDevice)
-			if err != nil {
-				glog.Errorf("Failed to unbind service from dummy interface, error: %v", err)
-			}
+			unbindIPAddr[svc.Address.String()] = true
+		}
+	}
+
+	for addr := range unbindIPAddr {
+		err := proxier.ipvs.UnbindVirtualServerAddress(addr, DefaultDummyDevice)
```

Comment:
Can we just check if IP address exist before doing unbind? Introducing an extra map probably is over-killed.

Reply:
There is no good way to check it. Do You mean add another function like "CheckVirtualServerAddressBind"  invoking `ip addr` ? But I think  it is more over-killed than this.

Reply:
No, I mean can we check the IP address exist in the function UnbindVirtualServerAddress()?

Reply:
Actually, service IPs can be more than 10K, and do you ever evaluate the memory consume of the unbindIPAddr map?

Reply:
There is only two ways to check in  UnbindVirtualServerAddress(), The first is judging the output info of `ip addr` command, and treating it as succeed, but I think it is not reliable. Another is remembing the address after unbind, It is almost the same as the way above.

Reply:
I found it is treated as already bound if `ip` returned 2 in `EnsureVirtualServerAddressBind`, but ref to the  manual of ip [link](http://man7.org/linux/man-pages/man8/ip.8.html#EXIT_STATUS), it just means an error was reported by the kernel.

Reply:
Sounds not a bad idea.

Reply:
I guess if you employ netlink, probably you don't need to rely on `ip addr` command output message.